src/main/java/com/ericsson/ei/subscription/InformSubscriber.java

```diff
@@ -123,168 +131,223 @@ public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson)
-            headers = addJenkinsCrumbData(headers, encoding, notificationMeta);
+        String encoding = Base64.getEncoder().encodeToString((username + ":" + password)
+                .getBytes());
+        headers.add("Authorization", "Basic " + encoding);
+        LOGGER.debug("Successfully added header for 'Authorization'");
 
-        } else {
-            LOGGER.error(
-                    "userName/password field in subscription is missing. Make sure both are provided for BASIC_AUTH.");
+        if (authType.equals(AUTHENTICATION_TYPE_BASIC_AUTH_JENKINS_CSRF)) {
+            headers = addJenkinsCrumbData(headers, encoding, notificationMeta);
         }
+
         return headers;
     }
 
     /**
-     * This method adds crumb to the headers if applicable.
+     * Returns a boolean indicating that authentication details was provided in the
+     * subscription
+     *
+     * @param authType
+     * @param username
+     * @param password
+     * @return
+     */
+    private boolean isAuthenticationDetailsProvided(String authType, String username,
+            String password) {
+        if (authType.isEmpty() || authType.equals(AUTHENTICATION_TYPE_NO_AUTH)) {
+            return false;
+        }
+
+        if (username.equals("") && password.equals("")) {
+            LOGGER.error("userName/password field in subscription is missing.");
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Adds crumb to the headers if applicable.
      *
      * @param headers
      * @param encoding
-     * @param notificationMeta
-     *     A String containing a URL
-     * @return headers
-     *     Headers containing Jenkins crumb data
+     * @param notificationMeta A String containing a URL
+     * @return headers Headers containing Jenkins crumb data
+     * @throws AuthenticationException
      */
-    private HttpHeaders addJenkinsCrumbData(HttpHeaders headers, String encoding, String notificationMeta) {
-        JsonNode jenkinsJsonCrumbData = fetchJenkinsCrumbIfAny(encoding, notificationMeta);
+    private HttpHeaders addJenkinsCrumbData(HttpHeaders headers, String encoding,
+            String notificationMeta) throws AuthenticationException {
+        LOGGER.info("Jenkins Crumb data is about to be fetched.");
+        JsonNode jenkinsJsonCrumbData = fetchJenkinsCrumb(encoding, notificationMeta);
         if (jenkinsJsonCrumbData != null) {
             String crumbKey = jenkinsJsonCrumbData.get("crumbRequestField").asText();
             String crumbValue = jenkinsJsonCrumbData.get("crumb").asText();
             headers.add(crumbKey, crumbValue);
-            LOGGER.debug("Successfully added header: " + String.format("'%s':'%s'", crumbKey, crumbValue));
+            LOGGER.info("Successfully added header: " + String.format("'%s':'%s'", crumbKey,
+                    crumbValue));
+        } else {
+            LOGGER.info("Failed to fetch Jenkins Crumb data!");
```

Comment:
Maybe clarify this log message a bit? Correct me if I'm wrong but this could happen when there is no CSRF enabled in Jenkins right? Then maybe rephrase the mesage so it is understood that there was no crumb to fetch but it's OK anyway.

Reply:
Or remove this else block entirely? Because the nested method fetchJenkinsCrumbIfAny logs all you need to know.

Reply:
May remove this logging as it is logged as a warning in enother function.